File: with-rust-postgres/src/update_data.rs
use dotenvy::dotenv;
use postgres::Client;
use openssl::ssl::{SslConnector, SslMethod};
use postgres_openssl::MakeTlsConnector;
use std::env;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    dotenv()?;
    let conn_string = env::var("DATABASE_URL")?;

    let builder = SslConnector::builder(SslMethod::tls())?;
    let connector = MakeTlsConnector::new(builder.build());

    let mut client = Client::connect(&conn_string, connector)?;
    println!("Connection established");

    // Update a data row in the table
    let updated_rows = client.execute(
        "UPDATE books SET in_stock = $1 WHERE title = $2",
        &[&true, &"Dune"],
    )?;
    
    if updated_rows > 0 {
        println!("Updated stock status for 'Dune'.");
    } else {
        println!("'Dune' not found or stock status already up to date.");
    }
    
    Ok(())
}

File: with-node-postgres-js/update_data.js
import 'dotenv/config';
import postgres from 'postgres';

const sql = postgres(process.env.DATABASE_URL, {
    ssl: 'require',
});

async function updateData() {
    try {
        console.log('Connection established');

        // Update a data row in the table
        await sql`UPDATE books SET in_stock = ${true} WHERE title = ${'Dune'}`;
        console.log("Updated stock status for 'Dune'.");
    } catch (err) {
        console.error('Connection failed.', err);
    } finally {
        sql.end();
    }
}

updateData();